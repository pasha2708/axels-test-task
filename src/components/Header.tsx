import React from 'react';

import Title from '../styled/components/Header';

const Header = () => <Title>Test App</Title>;

export default Header;
