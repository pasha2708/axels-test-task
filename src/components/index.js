export { default as Footer } from './Footer';
export { default as Gallery } from './Gallery';
export { default as Header } from './Header';
export { default as ModalWindow } from './ModalWindow';