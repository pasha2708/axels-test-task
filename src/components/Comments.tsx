import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useParams } from 'react-router-dom';

import EditComment from './EditComment';
import { BasicStateTypes } from '../redux/reducer';
import { sagaActions } from '../redux/sagaActions';
import {
	ButtonDelete,
	ButtonEdit,
	CommentsBlock,
	Comment,
} from '../styled/components/Comments';
import { dateFormat } from '../utils';

const Comments = () => {
	const { id: imageId } = useParams<{ id: string }>();
	const dispatch = useDispatch();
	const comments = useSelector(
		(state: BasicStateTypes) => state.fullImage.comments
	);
	const [editModal, setEditModal] = useState(false);
	const [commentData, setCommentData] = useState({});

	const deleteComment = (commentId: number) => {
		dispatch({
			type: sagaActions.DELETE_COMMENT,
			payload: { commentId, imageId },
		});
	};

	const openEditModal = (item: any) => {
		setEditModal(true);
		setCommentData(item);
	};

	return (
		<>
			<CommentsBlock>
				{comments &&
					comments.map((item) => (
						<Comment key={item.id}>
							<p>{dateFormat(item.date)}</p>
							<p>{item.text}</p>
							<ButtonDelete
								onClick={() => deleteComment(item.id)}
							/>
							<ButtonEdit onClick={() => openEditModal(item)} />
						</Comment>
					))}
			</CommentsBlock>
			{editModal && (
				<EditComment
					item={commentData}
					imageId={imageId}
					onClose={() => setEditModal(false)}
				/>
			)}
		</>
	);
};

export default Comments;
