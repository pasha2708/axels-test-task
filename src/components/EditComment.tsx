import React from 'react'
import { ButtonClose, Container, Overlay } from '../styled/components/ModalWindow'
import styled from 'styled-components'
import { validationSchema } from '../schemas'
import { useFormik } from 'formik';
import { useDispatch } from 'react-redux'
import { sagaActions } from '../redux/sagaActions'
import { StyledTextField } from './ModalWindow'
import { TextField, Button } from '@mui/material';
import { colors } from '../styled/vars';

const EditContainer = styled.div`
background-color: ${colors.primaryWhite};
	position: absolute;
	top: 50%;
	left: 50%;
	margin-right: -50%;
	transform: translate(-50%, -50%);
	padding: 2rem;
  width: 50%;
`

export const StyledForm = styled.div`
  display: flex;
  flex-direction: column;
  width: 100%;
`

const EditComment = (props: any) => {
  const dispatch = useDispatch();

  const formik: any = useFormik({
    initialValues: {
      author: props.item.author,
      text: props.item.text,
    },
    validationSchema: validationSchema,
    onSubmit: (values: any) => {
      formik.setSubmitting(true);
      dispatch({
        type: sagaActions.EDIT_COMMENT,
        payload: {
          imageId: props.imageId,
          id: props.item.id,
          date: new Date().getTime(),
          ...values
        },
      });
      formik.resetForm();
      formik.setSubmitting(false);
      props.onClose()
    },
  });
  return (
    <Overlay>
      <EditContainer>
        <ButtonClose onClick={props.onClose}>
          <span>&times;</span>
        </ButtonClose>
        <StyledForm>
          <StyledTextField>
            <TextField
              disabled
              fullWidth
              name='author'
              label='Name'
              value={formik.values.author}
            />
          </StyledTextField>
          <StyledTextField>
            <TextField
              required
              fullWidth
              name='text'
              type='text'
              label='Your comment'
              onChange={formik.handleChange}
              value={formik.values.text}
              error={formik.touched.text && Boolean(formik.errors.text)}
              helperText={(formik.touched.text && Boolean(formik.errors.text)) && formik.errors.text}
            />
          </StyledTextField>
          <Button
            fullWidth
            variant='contained'
            onClick={formik.handleSubmit}
            disabled={formik.isSubmitting}
          >
            Submit
          </Button>
        </StyledForm>
      </EditContainer>
    </Overlay>

  )
}

export default EditComment
