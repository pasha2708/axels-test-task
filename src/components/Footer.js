import React from 'react';
import { Line, FooterText, FooterDiv } from '../styled/components/Footer'

const Footer = () => (
  <FooterDiv>
    <Line></Line>
    <FooterText>Â© 2021</FooterText>
  </FooterDiv>
);

export default Footer;