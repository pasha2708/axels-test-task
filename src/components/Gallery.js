import React from 'react';
import { Switch, Route, Link } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import Row from 'react-bootstrap/Row';

import { ModalWindow } from '../components';
import { StyledCol, StyledContainer } from '../styled/components/Gallery';

import { GET_PREVIEW } from '../redux/ducks/gallery';

const Gallery = ({ data }) => {
  const dispatch = useDispatch();
  React.useEffect(() => {
    dispatch({type: GET_PREVIEW});
  }, [dispatch]);

  return (
    <StyledContainer>
      <Row>
        {data.images &&
          data.images.map((item) => (
           
              <StyledCol key={item.id} md={6} lg={4}>
                <Link to={`/images/${item.id}`}>
                  <img src={item.url} alt="preview" />
                </Link>
                <Switch>
                <Route path={`/images/${item.id}`}>
                  <ModalWindow id={item.id} data={data} />
                </Route>
              </Switch>
              </StyledCol>
              
            
          ))}
      </Row>
    </StyledContainer>
  );
};

export default Gallery;
