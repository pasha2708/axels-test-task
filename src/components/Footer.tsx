import React from 'react';

import { FooterText, FooterDiv } from '../styled/components/Footer'

const Footer = () => (
  <FooterDiv>
    <FooterText>{`Â© ${new Date().getFullYear()}`}</FooterText>
  </FooterDiv>
);

export default Footer;