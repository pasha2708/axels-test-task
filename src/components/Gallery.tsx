import React from 'react';
import { Routes, Route, Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';

import { ModalWindow } from '../components';
import { StyledCol, StyledContainer } from '../styled/components/Gallery';
import { sagaActions } from '../redux/sagaActions';
import { BasicStateTypes } from '../redux/reducer';
import styled from 'styled-components';
import { CircularProgress } from '@mui/material';

const GalleryStyle = styled.div`
	display: flex;
	justify-content: center;
	flex-wrap: wrap;
`;

const Gallery: React.FC = () => {
	const images = useSelector((state: BasicStateTypes) => state.images);
	const dispatch = useDispatch();
	React.useEffect(() => {
		dispatch({ type: sagaActions.FETCH_IMAGES });
	}, [dispatch]);

	return (
		<StyledContainer>
			<GalleryStyle>
				{!images ? (
					<CircularProgress />
				) : (
					images.map((item) => (
						<StyledCol key={item.id}>
							<Link to={`/images/${item.id}`}>
								<img src={item.url} alt='preview' />
							</Link>
						</StyledCol>
					))
				)}
				<Routes>
					<Route path={'/images/:id'} element={<ModalWindow />} />
				</Routes>
			</GalleryStyle>
		</StyledContainer>
	);
};

export default Gallery;
