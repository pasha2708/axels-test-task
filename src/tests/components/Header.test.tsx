import { shallow } from 'enzyme';

import { Header } from '../../components';

describe("Header component", () => {
  it("should render Header and take snapshot", () => {
    const component = shallow(<Header />);
    expect(component).toMatchSnapshot();
  });
});
