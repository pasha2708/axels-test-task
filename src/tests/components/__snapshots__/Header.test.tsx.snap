// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render Header and take snapshot 1`] = `
<styled.h1>
  Test App
</styled.h1>
`;
